crates/sc-consensus-subspace/src/lib.rs

```diff
@@ -1310,13 +1313,29 @@ where
 
         let import_result = self.inner.import_block(block, new_cache).await?;
         let (root_block_sender, root_block_receiver) = mpsc::channel(0);
+        let (block_import_throttling_sender, block_import_throttling_receiver) =
+            if self.executor_enabled {
+                let (sender, receiver) = mpsc::channel(0);
+                (Some(sender), Some(receiver))
+            } else {
+                (None, None)
+            };
 
         self.imported_block_notification_sender
             .notify(move || ImportedBlockNotification {
                 block_number,
                 root_block_sender,
+                block_import_throttling_sender,
             });
 
+        if let Some(mut throttling_receiver) = block_import_throttling_receiver {
+            if throttling_receiver.next().await.is_none() {
+                return Err(ConsensusError::ClientImport(
+                    "Block import throttling error".to_string(),
+                ));
```

Comment:
I don't see why this is an error, you should ignore the results, just wait until senders are dropped here.

Reply:
If there are multiple senders they will probably have their own subscription. In this case I don't expect anything like that.